sql/src/main/java/io/crate/analyze/MetaDataToASTNodeResolver.java

```diff
@@ -131,7 +131,7 @@ public Extractor(DocTableInfo tableInfo) {
                 }
 
                 List<ColumnConstraint<Expression>> constraints = new ArrayList<>();
-                if (!info.isNullable()) {
+                if (!tableInfo.primaryKey().contains(info.column()) && !info.isNullable()) {
```

Comment:
I'm not sure if we should add these special exceptions.

```
> create table tbl (x int primary key);
\d+ public.tbl
                                    Table "public.tbl"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 x      | integer |           | not null |         | plain   |              | 
Indexes:
    "tbl_pkey" PRIMARY KEY, btree (x)
Access method: heap


> create table tbl2 (x int, y int, primary key (x, y));
> \d+ public.tbl2
                                   Table "public.tbl2"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 x      | integer |           | not null |         | plain   |              | 
 y      | integer |           | not null |         | plain   |              | 
Indexes:
    "tbl2_pkey" PRIMARY KEY, btree (x, y)
Access method: heap
```

Not sure if that is too much of a change for a bugfix, but I really don't like adding exceptions like these. 
Too easy to forget to add it in a place and too hard to reason about why it's necessary in one place vs. not in another.

Reply:
as discussed, i will address nullable for primary keys separately from this analyzer fix.